### PR TITLE
GSoC proposal on Belle II Validation

### PR DESCRIPTION
**Pull request by tkuhr**

### CODE REVIEW

**Comment by tkuhr** [comment]
I'm not sure if the organization should be CERN instead of LMU as Belle II is a CERN recognized experiment.

**Comment by tkuhr** [comment]
Thanks, @hegner.
How do I assign reviewers?

**Comment by aflp91** [comment]
@tkuhr, you need to have a co-mentor and add your mails in `mentors.md` file.

In header project file you have to add `difficulty`, `duration` and `mentor_availability` informations.

**Comment by tkuhr** [comment]
Thanks for the quick feedback. I'm discussing with co-mentor candidates and will then update the PR.

**Comment by aflp91** [comment]
@agheata : I feel that the project title has to be in only one word? I have not a correct link to the `Belle II` project (as with `MadAnalysis 5` by the way) on my local site…

**Comment by tkuhr** [comment]
I'll then change the title to Belle2.
How could I test myself if it works?

**Comment by hegner** [comment]
You can follow this:
  https://hepsoftwarefoundation.org/jekyll-beginners.html#checking-the-results-of-your-contribution

**Comment by aflp91** [comment]
Hi @tkuhr ! This is how I'm testing … 

    git clone git@github.com:HSF/hsf.github.io.git hsf.github.io.git
    cd hsf.github.io.git
    git switch into-my-to-be-tested-branch
    docker run --rm -it \
      --volume="$PWD:/srv/jekyll" \
      --volume="$PWD/vendor/bundle:/usr/local/bundle" \
      -p 4000:4000 jekyll/jekyll:latest \
      jekyll serve

**Comment by agheata** [comment]
> I feel that the project title has to be in only one word? I do not have a suitable link to the Belle II project (as with MadAnalysis 5 by the way) on my local site…

It looks like the project is rendered correctly as is and the proposal is correctly associated, so no need to change the project name.

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+layout: gsoc_proposal
+project: Belle II
+year: 2022
+organization: LMU
+---
+
+## Description
+The international Belle II collaboration relies on custom software for the simulation, reconstruction, and analysis of the data recorded by the Belle II detector. The software was developed by hundreds of physicists over more than a decade. The development process will continue for several years to optimize algorithms for better physics performance or resource efficiency, to adjust the code to technological evolutions, or to implement new ideas.
+
+With such a large, diverse, and quickly changing development team the quality assurance of the software is a challenge. Belle II has implemented several mechanisms to monitor the software quality. An essential tool is the high-level validation. It runs algorithms on large datasets and produces plots of distributions of quantities that are relevant for physics analyses. The distributions are displayed on a web page and compared with previous versions and a reference. With this degradations can be quickly identified and fixed.
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:12]: degradations -> quality degradations ?(to be clear what degradations you are talking about)

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+While the validation system is used in production successfully, several items are know that could be improved:
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:14]: know -> known

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- A bookkeeping of known degradations
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:15]: quality degradations

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- A better referencing to individual plots
+- More information to encourage the reuse of artifacts
+- Improvements to make the root cause analysis of degradations easier
+
+This project seeks to enhance the user experience of the Belle II software validation page and make workflows more efficient by appropriate technical solutions. The estimated effort is 175 (or 300?) hours and the task is well suited for a student with experience in python and web programming. Knowledge of (particle) physics is not required.
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:20]: effort is 175 hours, but could be extended to 350 hours depending on ... (you need to explain)

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+## Task ideas
+ * Analyze the existing code base and understand the requirements.
+ * Discuss technical solutions with the mentors and implement them.
+ * Roll out new versions and collect user feedback.
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:25]: This is kind of very general, please try to add more specific items, like those enumerated in the introduction part

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+The expected result is an enhanced validation page that makes it easier to identify, keep track of, and resolve degradations introduced in the Belle II software.
+
+## Evaluation Task
+Interested students please contact Thomas (Thomas.Kuhr@lmu.de) for more details and an evaluation task.
+
+## Requirements
+ * Python, web programming, linux. Knowledge of cherrypy would be an advantage.
+
+## Mentors
+ * [Thomas Kuhr](mailto:Thomas.Kuhr@lmu.de)
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:37]: Affiliation needed; main mentor in bold, backup mentor mandatory (please check the existing proposals). The mentors have to be added to 2022/mentors.md

File: _gsocproposals/2022/proposal_Belle2Validation.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+title: Advanced Belle II Software Validation
+layout: gsoc_proposal
+project: Belle II
+year: 2022
+organization: LMU
```
Comment (agheata) [inline _gsocproposals/2022/proposal_Belle2Validation.md:6]: Project difficulty, duration and mentor availability required (please check the other projects)

**Review by agheata — CHANGES_REQUESTED** [review]

**Comment by agheata** [comment]
> It looks like the project is rendered correctly as is and the proposal is correctly associated, so no need to change the project name.

I was wrong, the project name has to be a single word indeed

File: gsoc/2022/mentors.md [hunk 8]
```diff
@@ -24,3 +24,5 @@ layout: plain
 * Cedric Sefron[Cedric Sefron](mailto:cedric.sefron@cern.ch) CERN
 * Mayank Sharma[Mayank Sharma](mailto:mayank.sharma@cern.ch) CERN
 * Anna Zaborowska[Anna Zaborowska](mailto:anna.zaborowska@cern.ch) CERN
+* Thomas Kuhr[Thomas Kuhr](mailto:Thomas.Kuhr@lmu.de) LMU
+* Nikolai Hartmann[Nikolai Hartmann](mailto:nikolai.hartmann@physik.uni-muenchen.de) LMU
```
Comment (hegner) [inline gsoc/2022/mentors.md:28]: mentors should be sorted alphabetically. 
fixing myself now.

Comment (tkuhr) [reply gsoc/2022/mentors.md]: Thanks!

**Pull request merged**